components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/api/APIGateway.java

```diff
@@ -92,4 +92,13 @@
      */
     void deleteEndpoint(Endpoint endpoint) throws GatewayException;
 
+    /**
+     * Add API state change to gateway
+     *
+     * @param api
+     * @param status
+     * @throws GatewayException
+     */
```

Comment:
param documentation and exception documentation is incomplete.
Please be more descriptive on method description so that others can understand what is the purpose of this method by looking  at the javadoc